ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -376,14 +402,60 @@ - (NSDictionary *)constantsToExport
     callback(@[[NSNumber numberWithBool:isPinOrFingerprintSet]]);
 }
 
+- (void)batteryLevelDidChange:(NSNotification *)notification
+{
+    float batteryLevel = [self.powerState[@"batteryLevel"] floatValue];
+    [self sendEventWithName:@"batteryLevelDidChange" body:@(batteryLevel)];
+
+    if (batteryLevel <= _lowBatteryThreshold) {
+        [self sendEventWithName:@"batteryLevelIsLow" body:@(batteryLevel)];
+    }
+}
+
+- (void)powerStateDidChange:(NSNotification *)notification
+{
+    [self sendEventWithName:@"powerStateDidChange" body:self.powerState];
+}
```

Comment:
When running this I saw a warning about emitting events without a listener. I originally thought adding a default listener was an idea to fix it but I read more, and the thing to do is to actually toggle whether you listen to system events and emit them to javascript based on whether you have listeners or not - the idea being that you don't do work (and spend device energy) if you don't have to. So you should subscribe/unsubscribe to the iOS service based on whether you have listeners. Explanation with code example here: https://facebook.github.io/react-native/docs/native-modules-ios.html#optimizing-for-zero-listeners

Reply:
perfect